Dockerfile

```diff
@@ -0,0 +1,95 @@
+# ##########
+# # COMMON #
+# ##########
+
+# The common image gives us all of the system packages upon which the stack
+# depends. We want to rebuild this as infrequently as possible, so as many
+# things as possible have been nailed down by checksums.
+FROM ruby:2.7.8 AS common
+
+# Create a non-root user; it's best practice not to run a container with the root user
+RUN useradd --create-home --shell /bin/bash codeandsupply
+
+# Require no interaction
+ENV DEBIAN_FRONTEND noninteractive
+
+# Keep the downloads together
+WORKDIR /tmp
+
+# A dependency of apt-fast-mirror, aria2c is a file transfer client that can fetch multiple files in parallel.
+ADD --checksum=sha256:8e7021c6d5e8f8240c9cc19482e0c8589540836747744724d86bf8af5a21f0e8 \
+    --chmod=644 \
+    --link \
+    https://github.com/aria2/aria2/releases/download/release-1.37.0/aria2-1.37.0.tar.gz aria2-1.37.0.tar.gz
+# Compile aria2c from source and install to /usr/local
+RUN tar xvzf aria2-1.37.0.tar.gz && \
+    cd aria2-1.37.0 && \
+    ./configure --prefix=/usr/local && \
+    make && make install
```

Comment:
**question:** Why not use what's in the distro's repos?

Reply:
This is frozen in time, the one from the distro requires an initial `apt-get update`